src/utils/deconflictChunk.ts

```diff
@@ -51,9 +51,18 @@ export function deconflictChunk(
 function deconflictImportsEsm(
 	usedNames: Set<string>,
 	imports: Set<Variable>,
-	_dependencies: Set<ExternalModule | Chunk>,
+	dependencies: Set<ExternalModule | Chunk>,
 	interop: boolean
 ) {
+	for (const chunkOrExternalModule of dependencies) {
```

Comment:
Moved this logic from `deconflictImportsOther` and only applied with the `index` variable name to account for `export * from './inner'` statements. However, I don't know what other implications this has! 🤔

Reply:
I'll have a look, thanks already for your work!

Reply:
Your proposed fix will only work for your example as the variable is only called "index" because your file is called "index". A general fix would be to run this for ALL dependencies, however this will cause a lot of test regressions (as you  may have noticed) due to a lot of unnecessary renames. The core of the problem is that reexporting a namespace is rendered (in `es.ts`) as importing the namespace as some variable and then exporting this variable, and this variable is not considered for deconflicting. A "nice" fix would be much more involved, but maybe this extends the scope of the bug fix. A simple hot "blanket" fix with only few regressions would be to  unconditionally deconflict the chunk variable names for all dependencies, but only do it if we are preserving modules. Maybe you could do that and add a comment what is fixed here and that this solution is not yet ideal (so that it is not forgotten)?

Reply:
The reason it is only necessary for preserveModules is that these kinds of reexports will not occur for other formats. Basically this line is responsible: https://github.com/rollup/rollup/blob/3ef6f073e4b54da2b2e61de5e3fd66f9f7352bca/src/Chunk.ts#L387

Reply:
Awesome, I will do that and can update the specs to see what that impacts.

Reply:
What I did was only apply this logic for `es` and `preserveModules: true` which lessens the impact so it doesn't impact current SystemJS behavior. Am I right in assuming this shouldn't really take effect for SystemJS? Otherwise it was affecting a lot more tests (56 vs. <10).